test cases/common/149 special characters/meson.build

```diff
@@ -0,0 +1,21 @@
+project('ninja special characters' ,'c')
+
+python = import('python3').find_python()
+
+ret = run_command(python,
+  ['check_quoting.py',
+   'newline=
+',
+   'dollar=$',
+   'colon=:',
+   'space= ',
+   'multi= :$
+ :$
+'])
```

Comment:
I doubled some stuff and mixed it around a bit; should definitely ensure there's no squashing going on.